File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/MainContract.java
package com.semisonfire.cloudgallery.ui.main;

import com.semisonfire.cloudgallery.ui.base.MvpPresenter;
import com.semisonfire.cloudgallery.ui.base.MvpView;

public interface MainContract {

    interface View extends MvpView {

    }

    interface Presenter<V extends View> extends MvpPresenter<V> {

    }
}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/MainContract.kt
package com.semisonfire.cloudgallery.ui.main

import com.semisonfire.cloudgallery.core.mvp.MvpPresenter
import com.semisonfire.cloudgallery.core.mvp.MvpView
import com.semisonfire.cloudgallery.data.remote.auth.Auth
import io.reactivex.Observable

interface MainContract {
  interface View : MvpView
  interface Presenter : MvpPresenter<View> {
    fun saveToken(token: String)
    fun getTokenListener(): Observable<Auth.AuthModel>
  }
}
